nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,315 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) mkIf types;
+  inherit (lib.lists) optionals;
+  inherit (lib.attrsets) optionalAttrs;
+  inherit (lib.strings) concatStringsSep escapeShellArg;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+
+  cfg = config.services.pr-tracker;
+in
+{
+  options.services.pr-tracker = {
+    enable = mkEnableOption "pr-tracker";
+
+    package = mkPackageOption pkgs "pr-tracker" { };
+
+    githubApiTokenFile = mkOption {
+      type = types.path;
+      example = "/run/secrets/gh-token";
+
+      description = ''
+        Path to a file containing your GitHub API token like so:
+
+        ```env
+        ghp_...
+        ```
+
+        ::: {.note}
+        The contents of this file will be the stdin of pr-tracker.
+        :::
+      '';
+    };
+
+    nixpkgsClone = {
+      cloneDir = mkOption {
+        type = types.path;
+        example = "/home/nixos/git/nixpkgs";
+
+        default = "${cfg.dataDir}/nixpkgs";
+        defaultText = "/var/lib/pr-tracker/nixpkgs";
+        description = ''
+          The path to the cloned nixpkgs pr-tracker will use.
+
+          ::: {.note}
+          If left as the default value this repo will automatically be cloned before
+          the pr-tracker server starts, otherwise you are responsible for ensuring
+          the directory exists with appropriate ownership and permissions.
+          :::
+        '';
+      };
+
+      remote = mkOption {
+        type = types.str;
+        example = "upstream";
+
+        default = "origin";
+        description = ''
+          The remote name in the repository corresponding to upstream Nixpkgs.
+        '';
+      };
+
+      managedByModule = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether you want this service to manage a clone of the nixpkgs
+          repo in `cfg.dataDir`.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        example = "15min";
+
+        default = "30min";
+        description = ''
+          How often to fetch nixpkgs if `cfg.nixpkgsClone.managedByModule` is true.
+        '';
+      };
+    };
+
+    userAgent = mkOption {
+      type = types.str;
+      example = "my pr-tracker";
+
+      description = ''
+        The User-Agent string to use when contacting the GitHub API.
+      '';
+    };
+
+    sourceUrl = mkOption {
+      type = types.str;
+      example = "https://github.com/me/my-pr-tracker-fork";
+
+      default = "https://git.qyliss.net/pr-tracker";
+      description = ''
+        The URL where users can download the program's source code.
+      '';
+    };
+
+    mountPath = mkOption {
+      type = with types; nullOr str;
+      example = "pr-tracker.html";
```

Comment:
Isn't a path normally starting with a slash `/`?

Reply:
For pr-tracker, it doesn't seem to be needed https://git.qyliss.net/pr-tracker/about/

Reply:
yay, 429 again 😅 

Is the / optional? I would prefer it to write it out as they are usually required in such places.

Reply:
Yeah, it's optional. Just tested it.